lib/init.js

```diff
@@ -0,0 +1,105 @@
+'use babel';
+
+// eslint-disable-next-line import/extensions, import/no-extraneous-dependencies
+import { CompositeDisposable } from 'atom';
+import * as helpers from 'atom-linter';
+import path from 'path';
+
+module.exports = {
+  config: {
+    executablePath: {
+      type: 'string',
+      title: 'dockerlint Executable Path',
+      default: path.join(__dirname, '..', 'node_modules', 'dockerlint', 'bin', 'dockerlint.js'),
+    },
+  },
+
+  activate() {
+    this.idleCallbacks = new Set();
+    let depsCallbackID;
+    const installLinterDockerDeps = () => {
+      this.idleCallbacks.delete(depsCallbackID);
+      if (!atom.inSpecMode()) {
+        require('atom-package-deps').install('linter-docker');
+      }
+    };
+    depsCallbackID = window.requestIdleCallback(installLinterDockerDeps);
+    this.idleCallbacks.add(depsCallbackID);
+
+    this.subscriptions = new CompositeDisposable();
+    this.subscriptions.add(
+      atom.config.observe('linter-docker.executablePath', (value) => {
+        this.executablePath = value;
+      }),
+    );
+  },
+
+  deactivate() {
+    this.idleCallbacks.forEach(callbackID => window.cancelIdleCallback(callbackID));
+    this.idleCallbacks.clear();
+    this.subscriptions.dispose();
+  },
+
+  provideLinter() {
+    return {
+      name: 'dockerlint',
+      grammarScopes: ['source.dockerfile'],
+      scope: 'file', // or 'project'
+      lintsOnChange: false, // must be false for scope: 'project'
+      lint: (textEditor) => {
+        const filePath = textEditor.getPath();
+        return helpers.execNode(this.executablePath, [filePath], { stream: 'stderr', throwOnStderr: false, allowEmptyStderr: true })
+          .then((output) => {
+            const lines = output.split('\n');
+            const patterns = [
+              {
+                // regex for "ERROR: ENV invalid format ENV_VARIABLE on line 2"
+                regex: /(WARN|ERROR):(.*) on line (\d+)/,
+                cb: m => ({ lineNo: m[3], excerpt: m[2], severity: m[1] }),
```

Comment:
Since we're doing arithmetic on it later, do a `Number.parseInt(m[3], 10)` on the `lineNo`.